modules/packages/Python.chpl

```diff
@@ -1902,6 +1961,87 @@ module Python {
     }
   }
 
+  /*
+    Represents a Python set. This provides a Chapel interface to Python sets,
+    where the Python interpreter owns the set.
+  */
+  class PySet: Value {
+    @chpldoc.nodoc
+    proc init(in interpreter: borrowed Interpreter,
+              in obj: PyObjectPtr, isOwned: bool = true) {
+      super.init(interpreter, obj, isOwned=isOwned);
+    }
+
+    /*
+      Get the size of the set. Equivalent to calling ``len(obj)`` in Python.
+
+      :returns: The size of the set.
+    */
+    proc size: int throws {
+      var size = PySequence_Size(this.getPyObject());
+      this.check();
+      return size;
+    }
+
+    /*
+      Add an item to the set.  Equivalent to calling ``obj.add(item)`` in
+      Python.
+
+      :arg item: The item to add to the set.
+     */
+    proc add(item: ?) throws {
+      PySet_Add(this.getPyObject(), interpreter.toPython(item));
+      this.check();
+    }
+
+    /*
+      Check if an item is in the set.  Equivalent to calling ``item in obj`` in
+      Python.
+
+      :arg item: The item to check for membership in the set.
+    */
+    proc contains(item: ?): bool throws {
+      var result = PySet_Contains(this.getPyObject(),
+                                  interpreter.toPython(item));
+      this.check();
+      return result: bool;
+    }
+
+    /*
+      Discard a specific item from the set.  Equivalent to calling
+      ``obj.discard(item)`` in Python.
+
+      :arg item: The item to discard from the set.
+    */
+    proc discard(item: ?) throws {
+      PySet_Discard(this.getPyObject(), interpreter.toPython(item));
+      this.check();
+    }
+
+    // TODO: how to handle pop?  Since Python sets can contain any type, I'm not
+    // sure what type to pass to `fromPython`
+    /*
+      Pop an arbitrary element from the set and return it.  Equivalent to
+      calling ``obj.pop()`` in Python.
+    */
+    /*
+    proc pop() throws {
+      var popped = PySet_Pop(this.getPyObject());
+      this.check();
+      return interpreter.fromPython(???, popped); // What to put here?
+    }
+    */
```

Comment:
I would have this be `proc pop(type T = owned Value)`. Then users can either specify a type if they know the type (they have a `set` of only `int`), or not specify a type and just work with the generic `Value` type.

Reply:
Ah right, good call!  I'd forgotten about the general Value type